router-tests/graphql_over_get_test.go

```diff
@@ -54,6 +55,97 @@ func TestOperationsOverGET(t *testing.T) {
 			require.Equal(t, `{"errors":[{"message":"Mutations can only be sent over HTTP POST"}]}`, res.Body)
 		})
 	})
+
+	t.Run("Query should be successful with custom path", func(t *testing.T) {
+		t.Parallel()
+
+		testenv.Run(t, &testenv.Config{
+			OverrideGraphQLPath: "/custom-graphql",
+		}, func(t *testing.T, xEnv *testenv.Environment) {
+			res, err := xEnv.MakeGraphQLRequestOverGET(testenv.GraphQLRequest{
+				OperationName: []byte(`Employees`),
+				Query:         `query Employees { employees { id } }`,
+			})
+			require.NoError(t, err)
+			require.Equal(t, http.StatusOK, res.Response.StatusCode)
+			require.Equal(t, `{"data":{"employees":[{"id":1},{"id":2},{"id":3},{"id":4},{"id":5},{"id":7},{"id":8},{"id":10},{"id":11},{"id":12}]}}`, res.Body)
+		})
+	})
+
+	t.Run("Mutation should not be allowed with custom path", func(t *testing.T) {
+		t.Parallel()
+
+		testenv.Run(t, &testenv.Config{
+			OverrideGraphQLPath: "/custom-graphql",
+		}, func(t *testing.T, xEnv *testenv.Environment) {
+			res, err := xEnv.MakeGraphQLRequestOverGET(testenv.GraphQLRequest{
+				OperationName: []byte(`updateEmployeeTag`),
+				Query:         "mutation updateEmployeeTag {\n  updateEmployeeTag(id: 10, tag: \"dd\") {\n    id\n  }\n}",
+			})
+			require.NoError(t, err)
+			require.Equal(t, http.StatusMethodNotAllowed, res.Response.StatusCode)
+			require.Equal(t, `{"errors":[{"message":"Mutations can only be sent over HTTP POST"}]}`, res.Body)
+		})
+	})
+
+	t.Run("Should return 404 for unknown path", func(t *testing.T) {
+		t.Parallel()
+
+		testenv.Run(t, &testenv.Config{
+			RouterOptions: []core.Option{
+				core.WithGraphQLPath("/custom-graphql"),
+			},
+		}, func(t *testing.T, xEnv *testenv.Environment) {
+			// Default path for creating requests is /graphql if not updated with `OverrideGraphQLPath`
+			res, err := xEnv.MakeGraphQLRequestOverGET(testenv.GraphQLRequest{
+				OperationName: []byte(`Employees`),
+				Query:         `query Employees { employees { id } }`,
+			})
+
+			require.NoError(t, err)
+			require.Equal(t, http.StatusNotFound, res.Response.StatusCode)
+		})
+	})
+
+	t.Run("Should not create wildcard for root path", func(t *testing.T) {
```

Comment:
Please also add a tests (I thought we would have that already) that the playground and the endpoint can be on the same path.

Reply:
Will do

Reply:
ty